opendevin/memory/condenser.py

```diff
@@ -0,0 +1,26 @@
+from opendevin.core.logger import opendevin_logger as logger
+from opendevin.llm.llm import LLM
+
+
+class MemoryCondenser:
+    def condense(self, summarize_prompt: str, llm: LLM):
+        """
+        Attempts to condense the monologue by using the llm
+
+        Parameters:
+        - llm (LLM): llm to be used for summarization
+
+        Raises:
+        - Exception: the same exception as it got from the llm or processing the response
+        """
+
+        try:
+            messages = [{'content': summarize_prompt, 'role': 'user'}]
```

Comment:
💭 This helper function looks quite generic except for the logging part. In other words, one can pass any prompt and everything would work even if the function call is a misuse.

Reply:
This is a good point--maybe the prompt should default to the existing one